plutus-tx/src/PlutusTx/AssocMap.hs

```diff
@@ -151,9 +160,12 @@ instance (Eq k, Semigroup v) => Monoid (Map k v) where
 instance (Pretty k, Pretty v) => Pretty (Map k v) where
   pretty (Map mp) = pretty mp
 
-{-# INLINEABLE fromList #-}
-fromList :: [(k, v)] -> Map k v
-fromList = Map
+{-# INLINEABLE unsafeFromList #-}
+-- | Unsafely create a 'Map' from a list of pairs. This should _only_ be applied to lists which
+-- have been checked to not contain duplicate pairs or duplicate keys, otherwise the 'Map' invariant
```

Comment:
- "duplicate pairs or duplicate keys" - is it not enough to just say "duplicate keys"?
- It may not be obvious what "the 'Map' invariant will be broken" means. I think it's worth elaborating a bit.